rag/llm/key_utils.py

```
@@ -31,4 +34,59 @@ def _normalize_replicate_key(key):
+        payload = key
+    elif isinstance(key, str):
+        try:
+            payload = json.loads(key)
+        except (json.JSONDecodeError, TypeError):
+            logging.warning(
+                "BaiduYiyan key is not valid JSON; expected a JSON object with 'yiyan_ak' and 'yiyan_sk' (see conf/models/baidu.json).",
+            )
+            raise ModelException(
+                'BaiduYiyan requires a JSON key with at least \'yiyan_ak\' and \'yiyan_sk\'. Example: {"yiyan_ak": "...", "yiyan_sk": "..."}. See conf/models/baidu.json for the model class.',
+                retryable=False,
+            )
```

Ruff (0.15.21)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)